code/modules/projectiles/guns/projectile/shotgun.dm

```diff
@@ -66,21 +67,49 @@
 
 // RIOT SHOTGUN //
 
+/datum/action/item_action/riot_shotgun/fold_stock
+	name = "(Un)fold Stock"
+	button_icon_state = "bolt_action"
+
+/datum/action/item_action/riot_shotgun/fold_stock/Trigger()
+	if(!Checks())
+		return
+
+	var/obj/item/weapon/gun/projectile/shotgun/riot/R = target
+	if(R.stock_folded)

```

Comment:
Move this into its own proc. Trigger should just do checks and then call a proc on the item.